src/overlay/SurveyManager.cpp

```diff
@@ -284,28 +298,44 @@ SurveyManager::processTopologyRequest(SurveyRequestMessage const& request) const
     response.ledgerNum = request.ledgerNum;
     response.surveyorPeerID = request.surveyorPeerID;
     response.surveyedPeerID = mApp.getConfig().NODE_SEED.getPublicKey();
-    response.commandType = SURVEY_TOPOLOGY;
+    response.commandType = request.commandType;
 
     SurveyResponseBody body;
-    body.type(SURVEY_TOPOLOGY);
-
-    auto& topologyBody = body.topologyResponseBody();
+    body.type(request.commandType);
 
     auto const& randomInboundPeers =
         mApp.getOverlayManager().getRandomInboundAuthenticatedPeers();
     auto const& randomOutboundPeers =
         mApp.getOverlayManager().getRandomOutboundAuthenticatedPeers();
 
-    populatePeerStats(randomInboundPeers, topologyBody.inboundPeers,
-                      mApp.getClock().now());
+    auto populateResponseBody = [&](auto& topologyBody) {
+        populatePeerStats(randomInboundPeers, topologyBody.inboundPeers,
+                          mApp.getClock().now());
+
+        populatePeerStats(randomOutboundPeers, topologyBody.outboundPeers,
+                          mApp.getClock().now());
 
-    populatePeerStats(randomOutboundPeers, topologyBody.outboundPeers,
-                      mApp.getClock().now());
+        topologyBody.totalInboundPeerCount =
+            static_cast<uint32_t>(randomInboundPeers.size());
+        topologyBody.totalOutboundPeerCount =
+            static_cast<uint32_t>(randomOutboundPeers.size());
+    };
 
-    topologyBody.totalInboundPeerCount =
-        static_cast<uint32_t>(randomInboundPeers.size());
-    topologyBody.totalOutboundPeerCount =
-        static_cast<uint32_t>(randomOutboundPeers.size());
+    bool extendedSurvey = request.commandType == SURVEY_TOPOLOGY_EXTENDED;
```

Comment:
updated way will be `bool extendedSurvey = body.type()` than couple with `commandType`